docs/source/format/CProtocol.rst

```diff
@@ -0,0 +1,546 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+=========================
+The Arrow C data protocol
+=========================
+
+Rationale
+=========
+
+Apache Arrow aims to be a universal in-memory format for the representation
+of tabular ("columnar") data, but some projects may face an undesirable
+choice between either depending on a fast-evolving dependency such as the
+Arrow C++ library, or having to implement adapters for data interchange
+(for example by reimplementing the Arrow IPC format, which is non-trivial
+and does not allow cross-runtime zero-copy).
+
+The Arrow C data protocol defines a very small, stable set of C definitions
+that can be easily *copied* in any project's source code and used for columnar
```

Comment:
copied from where? I don't see a *.c file that I would copy.

Reply:
From the C declarations which are bit later in the spec. Do you think the wording should be clearer?

Reply:
IMO "easily copied" doesn't mean cut-and-paste from code chunks embedded in a prose document. But either way, no, I wouldn't change this sentence here to clarify that you need to copy from this document, just make that clear below if that's the answer.